lambdas/log-redaction/src/redact-handler.ts

```diff
@@ -80,6 +57,81 @@ export class RedactHandler implements LambdaInterface {
     }
   }
 
+  private async saveRedactedLog(
+    redactLogGroup: string,
+    logStream: string,
+    logEvents: CloudWatchLogsDecodedData
+  ) {
+    logger.info("Putting redacted logs into " + redactLogGroup);
+
+    try {
+      const response: PutLogEventsCommandOutput = await cloudwatch.send(
+        new PutLogEventsCommand({
+          logGroupName: redactLogGroup,
+          logStreamName: logStream,
+          logEvents: logEvents.logEvents.map((event) => ({
+            id: event.id,
+            message: this.formatMessage(event.message),
+            timestamp: event.timestamp,
+            extractedFields: event.extractedFields,
+          })),
+        })
+      );
+      logger.info(JSON.stringify(response));
+    } catch (error) {
+      logger.error(`Error putting log events into ${redactLogGroup}: ${error}`);
+      throw error;
+    }
+  }
+
+  private async createLogStream(logStreamName: string, logGroupName: string) {
+    const query = await this.queryTable(logStreamName);
+
+    if (query.Count == 0) {
+      logger.info("Creating log stream " + logStreamName);
+
+      await cloudwatch.send(
+        new CreateLogStreamCommand({
+          logGroupName: logGroupName,
+          logStreamName: logStreamName,
+        })
+      );
+
+      await this.createLogStreamRecordInDB(logStreamName);
+    }
+  }
+
+  private async createLogStreamRecordInDB(logStream: string) {
+    await this.dynamodb.send(
+      new PutItemCommand({
+        TableName: logStreamTrackingTable,
+        Item: {
+          logStreamName: {
+            S: logStream,
+          },
+          expiryDate: {
+            N: (Math.floor(Date.now() / 1000) + 120 * 60).toString(),
+          },
+        },
+      })
+    );
+    logger.info("Added " + logStream + " to " + logStreamTrackingTable);
+  }
+
+  private async queryTable(logStream: string) {
```

Comment:
```suggestion
  private async logStreamExists(logStream: string) {
```
I would move the logic to check if the log stream exists to this function and return a bool, or create a separate function to check that count == 0